### PR TITLE
[OSDOCS-11570]: Adding docs about 4 and 5 node architecture

### PR DESCRIPTION
**Pull request by lahinson**
Version(s): 4.17

Issue: https://issues.redhat.com/browse/OSDOCS-11570

Link to docs preview: 
- Postinstallation cluster tasks: https://81631--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/cluster-tasks#post-install-etcd-tasks (added note)
- Node scaling for etcd: https://81631--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/recommended-performance-scale-practices/recommended-etcd-practices.html#etcd-node-scaling_recommended-etcd-practices (new concept topic about adding up to 5 nodes)
- Cluster control planes: https://81631--ocpdocs-pr.netlify.app/openshift-enterprise/latest/architecture/control-plane#defining-masters_control-plane (updated note)

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/architecture-machine-roles.adoc [hunk 1]
```diff
@@ -81,7 +81,7 @@ Extra controls apply to control plane machines to prevent you from deleting all
 [NOTE]
 ====
 ifndef::openshift-dedicated,openshift-rosa[]
-Exactly three control plane nodes must be used for all production deployments.
+Exactly three control-plane nodes must be used for all production deployments. However, on bare metal platforms, clusters that were installed by using the Assisted Installer can be scaled up to five control-plane nodes.
```
Comment (racedo) [inline modules/architecture-machine-roles.adoc:84]: The assisted installer installation method shouldn't be mandatory, however adding control plane nodes with the instructions from assisted seems to be the only way, does this make sense?

File: post_installation_configuration/cluster-tasks.adoc [hunk 2]
```diff
@@ -322,6 +322,11 @@ include::modules/nodes-cluster-enabling-features-cli.adoc[leveloffset=+2]
 == etcd tasks
 Back up etcd, enable or disable etcd encryption, or defragment etcd data.
 
+[NOTE]
+====
+If you deployed a bare-metal cluster by using the Assisted Installer, as part of your post-installation tasks, you can scale the cluster up to 5 nodes. For more information, see xref:../scalability_and_performance/recommended-performance-scale-practices/recommended-etcd-practices.adoc#etcd-node-scaling_recommended-etcd-practices[Node scaling for etcd].
```
Comment (racedo) [inline post_installation_configuration/cluster-tasks.adoc:327]: Similarly to the previous note, I think the requirement is just that the clusters are bare metal, regardless of how they were installed, as the resulting clusters from the different installation methods are all the same.

File: modules/architecture-machine-roles.adoc (hunk 1 above)
Comment (lahinson) [reply modules/architecture-machine-roles.adoc:84]: In that case, I'll remove the mention of Assisted Installer here and in the other note in the cluster tasks file.

File: modules/etcd-node-scaling.adoc [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+:_mod-docs-content-type: CONCEPT
+[id="etcd-node-scaling_{context}"]
+= Node scaling for etcd
+
+In general, clusters must have 3 control-plane nodes. However, if your cluster is installed on a bare metal platform, you can scale a cluster to up to 5 nodes as a post-installation task.
+
+For example, to scale from 3 to 4 nodes post installation, you add a host and install it as a control-plane node. Then, the etcd Operator scales accordingly to account for the additional node.
+
+Scaling to 4-node or 5-node clusters is available only on bare metal platforms for clusters that were installed by using the Assisted Installer.
```
Comment (racedo) [inline modules/etcd-node-scaling.adoc:13]: Not limited to Assisted Installer only, must be bare metal, resulting cluster from assisted installer, agent-based installer, UPI or IPI will be the same on day-2

Comment (lahinson) [reply modules/etcd-node-scaling.adoc:13]: Good catch. I'll update that.

**Comment by hasbro17** [comment]
LGTM from my side too.

File: post_installation_configuration/cluster-tasks.adoc [hunk 4]
```diff
@@ -322,6 +322,11 @@ include::modules/nodes-cluster-enabling-features-cli.adoc[leveloffset=+2]
 == etcd tasks
 Back up etcd, enable or disable etcd encryption, or defragment etcd data.
 
+[NOTE]
+====
+If you deployed a bare-metal cluster, as part of your post-installation tasks, you can scale the cluster up to 5 nodes. For more information, see xref:../scalability_and_performance/recommended-performance-scale-practices/recommended-etcd-practices.adoc#etcd-node-scaling_recommended-etcd-practices[Node scaling for etcd].
```
Comment (abrennan89) [inline post_installation_configuration/cluster-tasks.adoc:327]:
```suggestion
If you deployed a bare-metal cluster, you can scale the cluster up to 5 nodes as part of your post-installation tasks. For more information, see xref:../scalability_and_performance/recommended-performance-scale-practices/recommended-etcd-practices.adoc#etcd-node-scaling_recommended-etcd-practices[Node scaling for etcd].
```
Suggested tweak to make this more active

**Review by abrennan89 — COMMENTED** [review]
Just a few tweaks needed to be in line with some of our guidelines; overall great work @lahinson ! 🙂

File: modules/architecture-machine-roles.adoc [hunk 5]
```diff
@@ -81,7 +81,7 @@ Extra controls apply to control plane machines to prevent you from deleting all
 [NOTE]
 ====
 ifndef::openshift-dedicated,openshift-rosa[]
-Exactly three control plane nodes must be used for all production deployments.
+Exactly three control-plane nodes must be used for all production deployments. However, on bare metal platforms, clusters can be scaled up to five control-plane nodes.
```
[the remark this answers is not in the document]
Comment (lahinson) [reply modules/architecture-machine-roles.adoc:84]: Good to know. Thanks! Vale suggested that I hyphenate it. I'll keep the SSG guidance in mind going forward.

**Pull request merged**